drivers/soundwire/stream.c

```diff
@@ -1369,7 +1369,7 @@ static int _sdw_prepare_stream(struct sdw_stream_runtime *stream,
 			if (ret < 0) {
 				dev_err(bus->dev, "Compute params failed: %d\n",
 					ret);
-				return ret;
+				goto restore_params;
```

Comment:
the change is a good one, but I am wondering what happens in the 'aggregation' case with two links involved for a single stream. There's a possibility that one of the two links runs out of bandwith, so we would restore the bandwidth count on one link but not the other. This eror control would need a further update to be complete.